lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -228,6 +239,23 @@ def truncate_message(message)
         return message if message.bytesize <= MAX_BYTES
         message.byteslice(0...MAX_BYTES)
       end
+
+      # don't forward logs from a logger writing to a standard streams unless
+      # it happens to be the only logger in use
+      def mark_standard_streams_for_skipping
+        return unless @loggers.keys.size > 1
+
+        to_prune = []
+        @loggers.each do |logger, dev|
+          if dev === $stdout || dev === $stderr
+            logger.mark_skip_instrumenting
+            to_prune << logger
+          else
+            logger.clear_skip_instrumenting
```

Comment:
Not sure that it's right for us to clear if they aren't STDOUT/STDERR. At the very least, this raises the possibility of multi-threaded issues if the logger was in the midst of another call since we set the flags during our methods. Maybe just marking the skips is the right thing to do here?

Reply:
Great call. With [f32f947](https://github.com/newrelic/newrelic-ruby-agent/pull/1041/commits/f32f947a34b1c04d80c56bf029c494e21a21a632), non stdout/stderr loggers are no longer cleared for instrumenting.